extensions/src/main/java/org/pentaho/platform/plugin/services/metadata/SessionCachingMetadataDomainRepository.java

```diff
@@ -350,7 +350,7 @@ private void removeDomainFromIDCache( String domainId ) {
           if ( k instanceof String ) {
             String key = (String) k;
             if ( key != null && key.startsWith( DOMAIN_CACHE_KEY_PREDICATE ) ) {
-              Set<String> domainIds = (Set<String>) cacheManager.getFromRegionCache( CACHE_REGION, key );
+              Set<String> domainIds = new HashSet<String>((Set<String>) cacheManager.getFromRegionCache( CACHE_REGION, key ));
```

Comment:
Used here a new HashSet because all cache thread safe mechanisms are not eligible for hash objects once using a direct reference